### ISSUE TITLE
Kotlin version mismatch possible when using bindings from server

### ISSUE THREAD

**Issue by krzema12** [typesafegithub/github-workflows-kt#1756]
## Context

Kotlin compiler's version used by the bindings server is regularly updated, and generally tracks the newest one. Also, the bindings server is the default, recommended way of consuming action bindings.

## Problem

If someone uses a Kotlin version that is not compatible with the one that the bindings server used to compile the bindings, the compiler complains (TODO: add an error message here).

## Ideas

TODO

**Comment by LeoColman** [comment]
In Java we usually do

- Compile with Java 8
- Run on Java 250 no problem

Can't we use Kotlin 1.5 to compile things on the server and that would provide enough compatibility?
Naturally we don't need to go that low, but backwards compatibility should be possible

----
Maybe Kotlin has something like "CompileVersion"/"SourceVersion" same thing as Java?

**Comment by krzema12** [comment]
Yes, that's the idea I'd like to try out first, perhaps I'll reorder the ideas in the issue description.

**Comment by krzema12** [comment]
Asked on Slack: https://kotlinlang.slack.com/archives/C7L3JB43G/p1735652698014899
